File: c++/2125.cpp
// Link: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/description/
// Runtime: 76 ms
// Memory: 23.2 MB

class Solution {
public:
  int numberOfBeams(const std::vector<std::string> &bank) {
    int previousNumberOfLasers = 0, numberOfBeams = 0;
    for (const auto &row : bank) {
      int numberOfLasersInRow = 0;
      for (const auto &place : row)
        if (place == '1')
          numberOfLasersInRow++;

      if (numberOfLasersInRow > 0) {
        numberOfBeams += previousNumberOfLasers * numberOfLasersInRow;
        previousNumberOfLasers = numberOfLasersInRow;
      }
    }

    return numberOfBeams;
  }
};

File: python/2125.py
# Link: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/description/
# Runtime: 76 ms
# Memory: 19.38 MB


class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        previous_number_of_lasers, number_of_beams = 0, 0
        for row in bank:
            count_lasers = row.count("1")

            if count_lasers:
                number_of_beams += previous_number_of_lasers * count_lasers
                previous_number_of_lasers = count_lasers

        return number_of_beams
